addons/binding/org.openhab.binding.yamahareceiver/src/main/java/org/openhab/binding/yamahareceiver/internal/protocol/InputWithPresetControl.java

```diff
@@ -0,0 +1,38 @@
+/**
+ * Copyright (c) 2010-2017 by the respective copyright holders.
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.openhab.binding.yamahareceiver.internal.protocol;
+
+import java.io.IOException;
+import java.util.Set;
+
+import com.google.common.collect.Sets;
+import org.openhab.binding.yamahareceiver.YamahaReceiverBindingConstants;
+
+/**
+ * The preset control protocol interface
+ *
+ * @author David Graeff - Initial contribution
+ */
+
+public interface InputWithPresetControl extends IStateUpdateable {
+    /**
+     * List all inputs that are compatible with this kind of control
+     */
+    public static Set<String> supportedInputs = Sets.newHashSet(YamahaReceiverBindingConstants.INPUT_TUNER,
+            "NET_RADIO", "USB", "DOCK", "iPOD_USB", "PC",
```

Comment:
I added constants for the ones which I touched `INPUT_TUNER`, `INPUT_SPOTIFY`. and `INPUT_BLUETOOTH`. The rest is left out as I didn't want to alter too much files. On top of this I knew that @davidgraeff will change this when adding JSON protocol support. I wonder if the JSON protocol names inputs differently?!

However, I agree these should have been constants since the beginning :P